iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/ArgTreeImpl.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.docker.tree.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.sonar.iac.common.api.tree.Tree;
+import org.sonar.iac.docker.tree.api.ArgTree;
+import org.sonar.iac.docker.tree.api.KeyValuePairTree;
+import org.sonar.iac.docker.tree.api.SyntaxToken;
+
+public class ArgTreeImpl extends InstructionTreeImpl implements ArgTree {
+  private final List<KeyValuePairTree> keyValuePairs;
+
+  public ArgTreeImpl(SyntaxToken arg, List<KeyValuePairTree> argNames) {
+    super(arg);
+    this.keyValuePairs = argNames;
+  }
+
+  @Override
+  public List<KeyValuePairTree> keyValuePairs() {
+    return keyValuePairs;
+  }
+
+  @Override
+  public List<Tree> children() {
+    List<Tree> children = new ArrayList<>();
+    children.add(keyword);
+    keyValuePairs.forEach(keyValuePair -> children.addAll(keyValuePair.children()));
```

Comment:
I wouldn't expand all children like this. I would only add all keyValuePairs.

Reply:
Thanks for mentioning it, I was asking myself what were the expected children while implementing the ONBUILD instruction as it was even more obvious in this case.
I believe I did the same mistake for some other instruction, I will check again and either correct them if the PR is still open or else create a single ticket to correct the ones that are already merged.